### ISSUE TITLE
"Cannot drop air" error

### ISSUE THREAD

**Issue by LoneDev6** [PaperMC/Paper#2944]
### What behaviour is expected:
No error

### What behaviour is observed:
Error: https://hasteb.in/ifizuyec.rb

### Steps/models to reproduce:
Unknown.. probably:
- player join
- some entity spawn/despawn/die

### Plugin list:
https://hasteb.in/fatusida.md

### Paper build number:
This server is running Paper version git-Paper-89 (MC: 1.15.2) (Implementing API version 1.15.2-R0.1-SNAPSHOT)

### Anything else:
I have this problem since Paper/Spigot 1.14.4.
My request is to get an option to hide this message "Cannot drop air" as it spams in console every 5-15 minutes

**Comment by MiniDigger** [comment]
please post the actual error so we can tell you which of your 1 3 7 plugins does this.

**Comment by LoneDev6** [comment]
Thanks @MiniDigger , problem is that this is the full error.
No plugin is shown in the error since Spigot 1.14.4, so I'd likely have to remove one plugin at time and wait until I get the error again, but as you surely understand this is not something I can do while players are playing (and that problem occurs only when players play).
This used to happen also when I had just < 50 plugins

**Comment by LoneDev6** [comment]
Wops, I probably didn't press paste, I edited the message now

**Comment by MiniDigger** [comment]
Ah, this is different than I expected. 
Its trying to load ItemEntities, but somehow the ItemStack, which is stored in nbt and loaded via the readAdditionalSaveData/a method is either null or air.
Did you upgrade your world? Did you do a forceupgrade?

**Comment by LoneDev6** [comment]
Yep really strange.
Well, I had problem since 1.14.4 and my server world got generated on 1.14.4 stable releases, so outdated world is not likely to be the problem.
Only way I think this is fixable is to somehow hide that message as it's not breaking game or making server crash. Do you have any advice? I tried ConsoleSpamFix but didn't intercept this message, seems to be out of scope from the main logger? idk

**Comment by electronicboy** [comment]
> My request is to get an option to hide this message "Cannot drop air" as it spams in console every 5-15 minutes

no.

Something on your server is creating invalid data, either an underlying issue, or, you have command blocks or plugins creating invalid entities

**Comment by LoneDev6** [comment]
@electronicboy so you think the problem is 100% related to entities spawned not correctly?

**Comment by electronicboy** [comment]
That error is 100% due to inconsistent data, if you got that on a fresh 1.14 server especially, that is 100% either a bug in the server, or an issue with something in your setup; We would need actual replication info, ideally without 100+ plugins

**Comment by LoneDev6** [comment]
@electronicboy totally understandable.
Well.. do you have any advice on how to hide that message? It didn't cause any issue but it's annoying to have logs with this walltext, since I couldn't figure out what's causing this in a year

**Comment by Matrixchung** [comment]
> My request is to get an option to hide this message "Cannot drop air" as it spams in console every 5-15 minutes
Now here's a plugin you need:
[https://dev.bukkit.org/projects/console-spam-fix/files](url)

**Comment by LoneDev6** [comment]
Sadly doesn't work.
I mean, I already tried this before asking here for support, these are my rules but doesn't block the message:
- 'Caused by: java.lang.IllegalArgumentException: Cannot drop air'
- 'net.minecraft.server.v1_15_R1.ReportedException: Loading entity NBT'

**Comment by HexedHero** [comment]
Stop trying to hide the message, fix your actual issues not sweep them under the carpet.
Its 99% chance one of your 137 plugins which is a fucking insane amount.

**Comment by Matrixchung** [comment]
> Sadly doesn't work.
> I mean, I already tried this before asking here for support, these are my rules but doesn't block the message:
> 
> * 'Caused by: java.lang.IllegalArgumentException: Cannot drop air'
> * 'net.minecraft.server.v1_15_R1.ReportedException: Loading entity NBT'

You can only add 'Cannot drop air' to the rules instead of all of it.I have tested it on all of my servers and it worked fine.

**Comment by LoneDev6** [comment]
> 
> 
> Stop trying to hide the message, fix your actual issues not sweep them under the carpet.
> Its 99% chance one of your 137 plugins which is a fucking insane amount.

I cannot remove one plugin at time and wait days until error pops up.. since this problem occurs only when a lot of users are online I can't just make tests in production server, that's why I asked if it's a known issue.
Also I don't see how your comment is useful to anyone, you're just throwing insults and not giving a proper solution🤔 

> 
> 
> > Sadly doesn't work.
> > I mean, I already tried this before asking here for support, these are my rules but doesn't block the message:
> > 
> > * 'Caused by: java.lang.IllegalArgumentException: Cannot drop air'
> > * 'net.minecraft.server.v1_15_R1.ReportedException: Loading entity NBT'
> 
> You can only add 'Cannot drop air' to the rules instead of all of it.I have tested it on all of my servers and it worked fine.

Thanks a lot, that's the solution I was searching for

**Comment by LoneDev6** [comment]
> 
> 
> > Sadly doesn't work.
> > I mean, I already tried this before asking here for support, these are my rules but doesn't block the message:
> > 
> > * 'Caused by: java.lang.IllegalArgumentException: Cannot drop air'
> > * 'net.minecraft.server.v1_15_R1.ReportedException: Loading entity NBT'
> 
> You can only add 'Cannot drop air' to the rules instead of all of it.I have tested it on all of my servers and it worked fine.

Using consolespamfix didn't solve the issue, error still appears.
I set the rule like this:
- 'Cannot drop air'
Do you have any other advice? 
Maybe this error is in another "console channel", or something similar if exists.
Is it possible to natively hide some messages via spigot config?

**Comment by BillyGalbreath** [comment]
@LoneDev6 the error is caused by the server loading an entity that has air in it's equipment/inventory/hand/head slots. You just need to find which plugin is saving entities with air and remove/fix the plugin.

**Comment by LoneDev6** [comment]
I found out the issue, if anyone is interested.
It's about itemstacks on the ground with amount greater than their own material max stack amount.
As I found out seems Spigot somewhere is setting material to AIR if the ItemStack amount is over the max.

I will fix my stuff.

Anyway I don't think that a bypass can be directly added into Spigot/Paper as if I recall Minecraft uses byte to save amount, so values from -127 to 127 are allowed and this can't be changed. If we change the type from byte to int I think clientside would break
